src/main/cpp/src/hive_hash.cu

```diff
@@ -157,7 +157,7 @@ hive_hash_value_t __device__ inline hive_hash_function<cudf::timestamp_us>::oper
  * @tparam hash_function Hash functor to use for hashing elements. Must be hive_hash_function.
  * @tparam Nullate A cudf::nullate type describing whether to check for nulls.
  */
-template <template <typename> class hash_function, typename Nullate>
+template <template <typename> class hash_function, typename Nullate, int MAX_NESTED_LEN = 8>
```

Comment:
It is not a good idea to hardcode the nested length. It will crash when there are more than 8 nested layers.
If 8 is the upper limitation, we need to check the length and throw an exception before running into this row hasher, and note this limitation in the doc of this function/class.
Another feasible way is we can calculate this length from the schema (simliar as what `check_hash_compatibility` does) and allocate a GPU stack (e.g. `rmm::device_vector`, or rmm::device_buffer) and pass the stack into the hive_device_row_hasher.

Reply:
>Another feasible way is we can calculate this length from the schema (you can get the schema from the input table view) and allocate a GPU stack

`get-json-object` is also using a constant depth. I remember that `allocate a GPU stack` will impact performance.
I recommand that check the max length in CPU cpp code before call into this kernel and throw an exception if length exceeds 8. In plugin Java code, add fallback logic.

Reply:
The GPU stack (actually a GPU memory) is allocated with a fixed size calculated from the schema, just like allocating an output column (a GPU memory inside). So it should have no big impact on perf, I think.